pkg/node/node.go

```diff
@@ -278,37 +278,58 @@ func NewBee(
 		return nil, fmt.Errorf("compute overlay address: %w", err)
 	}
 
-	if nonceExists && o.TargetNeighborhood != "" {
-		logger.Warning("an overlay has already been created before, skipping targeting the selected neighborhood")
+	targetNeighborhood := o.TargetNeighborhood
+	if targetNeighborhood == "" && !nonceExists && o.NeighborhoodSuggester != "" {
+		logger.Info("fetching target neighborhood from suggester", "url", o.NeighborhoodSuggester)
+		targetNeighborhood, err = nbhdutil.FetchNeighborhood(&http.Client{}, o.NeighborhoodSuggester)
+		if err != nil {
+			return nil, fmt.Errorf("neighborhood suggestion: %w", err)
+		}
 	}
 
-	if !nonceExists {
-		// mine the overlay
-		targetNeighborhood := o.TargetNeighborhood
-		if o.TargetNeighborhood == "" && o.NeighborhoodSuggester != "" {
-			logger.Info("fetching target neighborhood from suggester", "url", o.NeighborhoodSuggester)
-			targetNeighborhood, err = nbhdutil.FetchNeighborhood(&http.Client{}, o.NeighborhoodSuggester)
-			if err != nil {
-				return nil, fmt.Errorf("neighborhood suggestion: %w", err)
-			}
+	if targetNeighborhood != "" {
+		neighborhood, err := swarm.ParseBitStrAddress(targetNeighborhood)
+		if err != nil {
+			return nil, fmt.Errorf("invalid neighborhood. %s", targetNeighborhood)
 		}
 
-		if targetNeighborhood != "" {
+		if swarm.Proximity(swarmAddress.Bytes(), neighborhood.Bytes()) < uint8(len(targetNeighborhood)) {
+			if nonceExists {
+				logger.Info("Override nonce %d and clean state for neighborhood %s", nonce, targetNeighborhood)
+				logger.Warning("you have another 10 seconds to change your mind and kill this process with CTRL-C...")
+				time.Sleep(10 * time.Second)
+
+				const (
+					localstore = "localstore"
+					kademlia   = "kademlia-metrics"
+					statestore = "statestore"
```

Comment:
TLDR The cleanest way is instead to remove directories, it would be better for these services to implement its own internal nuke methods (that handle file descriptors and internal storage reinitalizations) and call them here, instead to remove their files.

Statestore directory is removed, but a few lines bellow statestore is passed to the setOverlay function. I would assume that the statestore is no longer reliable after its files are removed. Special attention should be made to ensure when interfering with storage internals outside of the services that manage their state. In this case, statestore must be reinitialised (and passed again to all already initialized services that use statestore) or if possible initilizad after the data on disk is removed. Handling nuke on node startup is different than in a separate nuke cli command as the node continues to run.

Nuking buy removing data directories is a questionable operation as these paths are constructed locally by assumption that they will not change in other places of the code where services that store data in them are initialized. I know that this is a mistake taken from the nuke cli command, but it would be good to make new code better.

Reply:
good point, I reinited statestore after the wipe as it is needed to fetch `nonce` before wipe.

Regarding the nuking, no service uses these databases before this point, all affected DB initializations happen in this function. It should have a separate `clearDb` method with options for sure if some of the data was required to be kept (please let me know if there is any). 
so basically, this is the only place where databases should managed as files, shouldn't it?

Reply:
Thank you for fixing it.

The statestore as a general key/value store, is a tricky to clean. I believe that this is a good time to plan something like a consolidation of all keys (or key prefixes) that are stored. It would be good to remove only the required keys.

I would say that it would even better to have a Nuke function on the databases to cleanup internal storage, in this case leveldb and not to worry about the file management. This would require a few more changes, but I believe that would result in more reliable code, not to worry about the files, just databases as abstraction. This can be an improvement later.

Reply:
the statestore actually has a nuke function but it preserves keys related to maintaing the overlay addresses, it may be improved to remove those entries when called upon.

Reply:
It is much better to adapt StateStoreAdapter.Nuke for cleaning. I think that it would be the safer way, and maybe to have the list of preserved prefixes a bit more transparent.